src/view/renderer.js

```diff
@@ -243,6 +236,67 @@ export default class Renderer {
 		this.markedChildren.clear();
 	}
 
+	/**
+	 * Updates viewElement children mappings. Children which were replaced in the view structure by the similar
+	 * element (same tag name) are treated as 'replaced'. Their mappings are rebind to the corresponding,
+	 * existing DOM element so they will not be replaced by a new DOM element during rendering.
+	 *
+	 * @private
+	 * @param {module:engine/view/node~Node} viewElement The view element which children mappings will be updated.
+	 */
+	_updateChildrenMappings( viewElement ) {
+		// We do not perform any operations on DOM here so there is no need to bind view element or convert its children.
+		const diff = this._diffElementChildren( viewElement, { bind: false, withChildren: false } );
+
+		if ( diff ) {
+			const actions = this._findReplaceActions( diff.actions, diff.actualDomChildren, diff.expectedDomChildren );
+
+			if ( actions.indexOf( 'replace' ) !== -1 ) {
+				const counter = { equal: 0, insert: 0, delete: 0 };
+				for ( const action of actions ) {
+					if ( action === 'replace' ) {
+						const insertIndex = counter.equal + counter.insert;
+						const deleteIndex = counter.equal + counter.delete;
+						const viewChild = viewElement.getChild( insertIndex );
+
+						// The 'uiElement' is a special one and its children are not stored in a view (#799),
+						// so we cannot use it with replacing flow (since it uses view children during rendering
+						// which will always result in rendering empty element).
+						if ( viewChild && !viewChild.is( 'uiElement' ) ) {
```

Comment:
This function grew a bit long. So, perhaps contents of this `if()` would make a reasonable function itself. It only accepts 3 params – the 3 vars that you define above. That's why I found it a good place to split the code a bit.